pkg/machines/reducers.es6

```diff
@@ -160,13 +160,22 @@ function vms(state, action) {
         return replaceVm({ state, updatedVm, index: indexedVm.index });
     }
     case UNDEFINE_VM: {
-        return state
-                .filter(vm => (action.connectionName !== vm.connectionName || action.name != vm.name ||
-                    (action.transientOnly && vm.persistent)));
+        if (action.id)
+            return state
+                    .filter(vm => (action.connectionName !== vm.connectionName || action.id != vm.id ||
+                        (action.transientOnly && vm.persistent)));
+        else
+            return state
+                    .filter(vm => (action.connectionName !== vm.connectionName || action.name != vm.name ||
+                        (action.transientOnly && vm.persistent)));
```

Comment:
.. the above (unifying name and id) should also avoid these nasty case handling, i. e. the backend specifics leaking into the abstract reducers.

Reply:
Resolving as it's the same as what we discussed above.